### ISSUE TITLE
FP S2259: Suppress warnings for lifted operator results in null value in value type comparison

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6117]
We have lots of FPs with the new engine whenever a nullable value type is compared/lifted to a non-nullable value type:

```cs
if (nullableInt == 0) { // nullableInt not null here }
if (arg?.Length > 3) { // arg not null here }
if (arg?.IsEmpty ?? true) { // arg not null here }
```

We track the work on this in [MMF-2401](https://sonarsource.atlassian.net/browse/MMF-2401) (internal link).
As a workaround and to reduce noise, we should suppress any warnings, where nullable value types are involved.

See also #4250
